Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,167 @@
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this.get('blocks').forEach(function (block) {
+                fragment.append(block.render().get('container'));
+            });
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided list of blocks by priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Array} list of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            return blocks.sort(function (a, b) {
+                return a.get('priority') > b.get('priority');
+            });
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.DashboardBlockBaseView} dashboard block view instance
+         * @return {Array} the list of blocks
+         */
+        addBlock: function (view) {
+            var blocks = this.get('blocks'),
+                identifier = view.get('identifier');
+
+            if (!identifier || !(view instanceof Y.eZ.DashboardBlockBaseView)) {
+                return blocks;
+            }
+
+            if (this._findBlock(identifier)) {
+                this.removeBlock(identifier);
+            }
+
+            view.addTarget(this);
+            view.set('active', this.get('active'));
+
+            blocks.push(view);
+
+            blocks = this._sortBlocks(blocks);
+
+            this._set('blocks', blocks);
+
+            return blocks;
+        },
+
+        /**
+         * Finds block with a given identifier
+         *
+         * @method _findBlocks
+         * @param identifier {String} block identifier
+         * @return {Y.eZ.DashboardBlockBaseView} a block view
+         */
+        _findBlock: function (identifier) {
+            return this.get('blocks').filter(function (block) {
+                return block.get('identifier') === identifier;
+            })[0];
+        },
+
+        /**
+         * Removes a block from the dashboard
+         *
+         * @method removeBlock
+         * @param identifier {String} block identifier
+         * @return {Array} the list of blocks
+         */
+        removeBlock: function (identifier) {
+            var blocks = this.get('blocks'),
+                existingBlock = this._findBlock(identifier);
+
+            existingBlock.removeTarget(this);
+            existingBlock.destroy({remove: true});
+
+            blocks.splice(blocks.indexOf(existingBlock), 1);
+
+            this._set('blocks', blocks);
+
+            return blocks;
+        }
+    }, {
+        ATTRS: {
+            /**
+             * An array dashboard blocks view instances
+             *
+             * @attribute blocks
+             * @type Array
```

Comment:
missing the `@readOnly` tag and maybe a warning in the description saying that to manipulate, you have to use `addBlock` and `removeBlock`.